src/Physics/HEDIS/XSection/HEDISStrucFunc.h

```diff
@@ -0,0 +1,267 @@
+    std::getline (is,saux); a.Vcd=atof(saux.c_str());
+    std::getline (is,saux); a.Vcs=atof(saux.c_str());
+    std::getline (is,saux); a.Vcb=atof(saux.c_str());
+    std::getline (is,saux); a.Vtd=atof(saux.c_str());
+    std::getline (is,saux); a.Vts=atof(saux.c_str());
+    std::getline (is,saux); a.Vtb=atof(saux.c_str());
+    return is;
+
+}
+
+  inline std::ostream & operator<< (std::ostream& os, const SF_info& a) {
+
+    return os << '\n'
+              << "#--------------------------------------------------------------------------------"    << '\n'
+              << "# Metafile that stores information used to generate Structure Functions for HEDIS"   << '\n'
+              << "#--------------------------------------------------------------------------------"    << '\n'
+              << "# LHAPDF set"    << '\n'
+              << a.LHAPDFset     << '\n'
+              << "# LHAPDF member" << '\n'
+              << a.LHAPDFmember  << '\n'
+              << "# NLO"        << '\n' 
+              << a.IsNLO         << '\n' 
+              << "# Mass Scheme"       << '\n'
+              << a.Scheme        << '\n'
+              << "# Quark threshold"      << '\n'
+              << a.QrkThrs       << '\n'
+              << "# NX"       << '\n'
+              << a.NGridX        << '\n'
+              << "# NQ2"      << '\n'
+              << a.NGridQ2       << '\n'
+              << "# Xmin"     << '\n'
+              << a.XGridMin      << '\n'
+              << "# Q2min"    << '\n'
+              << a.Q2GridMin     << '\n'
+              << "# Q2max"    << '\n'
+              << a.Q2GridMax     << '\n'
+              << "# Mass W"        << '\n'
+              << a.MassW         << '\n'
+              << "# Mass Z"        << '\n'
+              << a.MassZ         << '\n'
+              << "# Rho"          << '\n'
+              << a.Rho           << '\n'
+              << "# Sin2ThW"      << '\n'
+              << a.Sin2ThW       << '\n'
+              << "# CKM"          << '\n'
+              << a.Vud           << '\n'
+              << a.Vus           << '\n'
+              << a.Vub           << '\n'
+              << a.Vcd           << '\n'
+              << a.Vcs           << '\n'
+              << a.Vcb           << '\n'
+              << a.Vtd           << '\n'
+              << a.Vts           << '\n'
+              << a.Vtb           << '\n';
+
+  }
+
+  struct SF_xQ2 {
+
+    double F1;
+    double F2;
+    double F3;
+
+  };
+
+  class HEDISStrucFunc
+  {
+    public:
+
+      // ................................................................
+      // HEDIS structure functions type
+      //
+
+      typedef enum HEDISStrucFuncType {  
+        kMHTUndefined = 0,
+        kSFT1, 
+        kSFT2, 
+        kSFT3, 
+        kSFnumber, 
+      } 
+      HEDISStrucFuncType_t;
+
+      // ................................................................
+      // HEDIS form factor type
+      //
+
+      class HEDISStrucFuncTable 
```

Comment:
Do we need a singleton for this?

Reply:
Another way to express my doubts here is: why can we not have in the memory two different tables?